osbuild/objectstore.py

```diff
@@ -26,6 +26,32 @@ def suppress_oserror(*errnos):
             raise e
 
 
+class TreeObject:
```

Comment:
Yeah, you are right. Kinda mixed the object representing a temporary tree and and an object representing a tree stored in the object store (temporary or not) together, because it was convenient. Reworked it to just represent a tree (living in the object store) for now. I think that is better now.